File: C++/9.Palindrome_Number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        vector<int> arr;
        int n = 0;
        if(0<=x && x<=9){
            return true;
        }
        if(x<0){
            return false;
        }
        
        int y = x;
        while(y){
            int num = y%10;
            y = y/10;
            arr.push_back(num);
            // cout<<num<<"\n";
            n++;
        }
        
        int i,j;
        i=0;
        j=n-1;
        
        while(i<j){
            // cout<<arr[i]<<" "<<arr[j]<<"\n";
            if(arr[i]!=arr[j]){
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
};

File: Python/9.Palindrome Number.py
#This program returns true if the number is a palindrome, else it returns false

class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        #All negative numbers reutrn false for Palindrome
        if x < 0:
            return False
        
        a = x
        sum = 0
        while a > 0 :
            r = a % 10
            sum = sum*10 + r
            a = a // 10
        #If reversed number is same as orignal, returns True
        if sum==x:
            return True
        
        #else returns false
        else:
            return False
